### PR TITLE
Add interactive mode prompt to create-react-on-rails-app

### PR DESCRIPTION
**Pull request by justin808**
## Summary
- When running `npx create-react-on-rails-app my-app` without `--pro` or `--rsc`, the CLI now shows an interactive prompt to choose between Standard, Pro, and RSC modes (default: RSC recommended)
- Non-interactive environments (CI, pipes) fall back to standard mode automatically for backward compatibility
- Explicit `--pro` or `--rsc` flags skip the prompt entirely

## Test plan
- [x] All 85 existing tests pass (no regressions)
- [x] 7 new tests for `promptForMode()` covering all choices, default, invalid input, and whitespace trimming
- [x] ESLint, Prettier, and TypeScript type-check all pass
- [ ] Manual test: `npx create-react-on-rails-app test-app` shows mode prompt
- [ ] Manual test: `npx create-react-on-rails-app test-app --rsc` skips prompt
- [ ] Manual test: `echo | npx create-react-on-rails-app test-app` uses standard mode (non-TTY)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes the default interactive behavior of the scaffolding CLI by adding a mode prompt that defaults to RSC, which can alter generated app setup and introduce a Pro gem dependency if accepted. Non-interactive runs keep a standard-mode fallback, reducing CI risk.
> 
> **Overview**
> `create-react-on-rails-app` now prompts (in TTY environments) to choose **Standard**, **Pro**, or **RSC** mode when no mode flag is provided, defaulting to *RSC*; explicit `--standard`, `--pro`, or `--rsc` skips the prompt, and non-TTY runs automatically fall back to standard mode.
> 
> This introduces a new `promptForMode()` implementation with clean cancel handling, updates CLI help text and OSS docs to reflect the new mode-selection flow, and adds Jest coverage for both the prompt behavior and TTY/non-TTY branching.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bf4e81cd76a8ada9031c3bfbe6abec266de6fd39. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Interactive mode selection during app creation when no mode flag is provided; choose Standard, Pro, or RSC. --pro or --rsc still skips the prompt.

* **Behavior Changes**
  * In non-interactive environments the CLI skips prompting and defaults to Standard while logging info. Empty or invalid interactive input defaults to RSC. Canceling the prompt exits cleanly (no error), unexpected failures surface as errors.

* **Documentation**
  * Help text and docs updated to describe prompting, defaults, and how to skip it with flags.

* **Tests**
  * Added tests covering the interactive prompt, inputs, defaults, cancellation, and non-interactive branching.

### CODE REVIEW

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      const key = answer.trim() || DEFAULT_KEY;
+      const selected = MODES.find((m) => m.key === key);
+      if (!selected) {
+        console.log(chalk.yellow(`Invalid choice "${key}", defaulting to RSC.`));
+        resolve({ pro: false, rsc: true });
+        return;
+      }
+      resolve({ pro: selected.pro, rsc: selected.rsc });
+    });
+  });
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:77]: Fixed in d0356bc5d. Added `rl.on('SIGINT', ...)` to reject the promise and `rl.once('close', ...)` to resolve with RSC default on EOF. Both paths are now covered by tests.

File: packages/create-react-on-rails-app/src/index.ts [hunk 2]
```diff
@@ -44,6 +40,27 @@ function run(appName: string, rawOpts: Record<string, unknown>): void {
     process.exit(1);
   }
 
+  // When no mode flag is explicitly passed, prompt interactively (TTY only).
+  // Non-interactive environments (CI, pipes) fall back to standard mode.
+  const modeExplicit = rawOpts.pro !== undefined || rawOpts.rsc !== undefined;
+  if (!modeExplicit) {
+    if (process.stdin.isTTY) {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Fixed in d0356bc5d. Now gating on both `process.stdin.isTTY && process.stdout.isTTY` before entering interactive mode.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:78]: Fixed in d0356bc5d — added `rl.once('close', ...)` handler that resolves with RSC default when stdin closes before the question callback fires. Test added.

[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:78]: Fixed in d0356bc5d — same approach: `rl.once('close', ...)` resolves with RSC default on EOF. The `answered` guard prevents double-resolution on the happy path.

File: packages/create-react-on-rails-app/src/index.ts [hunk 4]
```diff
@@ -44,6 +40,27 @@ function run(appName: string, rawOpts: Record<string, unknown>): void {
     process.exit(1);
   }
 
+  // When no mode flag is explicitly passed, prompt interactively (TTY only).
+  // Non-interactive environments (CI, pipes) fall back to standard mode.
+  const modeExplicit = rawOpts.pro !== undefined || rawOpts.rsc !== undefined;
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Declining — the variable name `modeExplicit` plus the existing inline comment on lines 43-44 already convey the intent. Adding a second comment explaining the Commander undefined-vs-false behavior would be over-documenting an implementation detail.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  {
+    key: '1',
+    label: 'Standard',
+    desc: 'Open-source React on Rails with SSR',
+    pro: false,
+    rsc: false,
+  },
+  {
+    key: '2',
+    label: 'Pro',
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:21]: Declining for this PR — the current description matches the terminology used in our docs and other CLI output. Wording refinements can be revisited separately if users report confusion.

File: packages/create-react-on-rails-app/src/index.ts [hunk 6]
```diff
@@ -145,13 +165,13 @@ The generated app includes one git commit per logical setup step.`,
...
-      run(appName, opts);
+      await run(appName, opts);
     } catch (error) {
       logError(error instanceof Error ? error.message : String(error));
       process.exit(1);
     }
   });
 
-program.parse();
+void program.parseAsync();
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Declining — the `.action()` handler already wraps `run()` in try/catch (lines 168-174), and Commander itself calls `process.exit()` for argument errors. The `void` pattern is idiomatic for top-level async calls where the rejection path is already handled.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    let answered = false;
+
+    rl.on('SIGINT', () => {
+      rl.close();
+      reject(new Error('Prompt cancelled by user (SIGINT)'));
+    });
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:50]: Addressed in 2a47389bb. We now set `answered = true` before `rl.close()` in the SIGINT handler, so the close fallback cannot resolve first. The SIGINT path now deterministically rejects with an explicit cancellation error.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    rl.once('close', () => {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:52]: Addressed in 2a47389bb. We now set `answered = true` before `rl.close()` in the SIGINT handler, so the close fallback cannot resolve first. The SIGINT path now deterministically rejects with an explicit cancellation error.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    let answered = false;
+
+    rl.on('SIGINT', () => {
+      rl.close();
+      reject(new Error('Prompt cancelled by user (SIGINT)'));
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts]: Addressed in 2a47389bb. We now set `answered = true` before `rl.close()` in the SIGINT handler, so the close fallback cannot resolve first. The SIGINT path now deterministically rejects with an explicit cancellation error.

File: packages/create-react-on-rails-app/src/prompt.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:50]: Addressed in 2a47389bb. We now set `answered = true` before `rl.close()` in the SIGINT handler, so the close fallback cannot resolve first. The SIGINT path now deterministically rejects with an explicit cancellation error.

File: packages/create-react-on-rails-app/tests/prompt.test.ts [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    expect(result).toEqual({ pro: false, rsc: false });
+  });
+
+  it('rejects when user presses Ctrl+C (SIGINT)', async () => {
+    mockRl.question.mockImplementation(() => {});
+    const promise = promptForMode();
+    for (const handler of eventHandlers['SIGINT'] ?? []) handler();
+    await expect(promise).rejects.toThrow('Prompt cancelled by user (SIGINT)');
+    expect(mockRl.close).toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/prompt.test.ts:94]: Addressed in 2a47389bb. We now set `answered = true` before `rl.close()` in the SIGINT handler, so the close fallback cannot resolve first. The SIGINT path now deterministically rejects with an explicit cancellation error.

File: packages/create-react-on-rails-app/tests/prompt.test.ts [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+  it('defaults to rsc when stdin closes (EOF/Ctrl+D)', async () => {
+    mockRl.question.mockImplementation(() => {});
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/prompt.test.ts]: Addressed in 2a47389bb. The prompt test mock now makes `close()` fire registered `close` handlers, so this readline settlement race is exercised in tests instead of being masked by the mock.

File: packages/create-react-on-rails-app/src/index.ts [hunk 12]
```diff
@@ -145,13 +165,13 @@ The generated app includes one git commit per logical setup step.`,
...
   )
-  .action((appName: string, opts: Record<string, unknown>) => {
+  .action(async (appName: string, opts: Record<string, unknown>) => {
     try {
-      run(appName, opts);
+      await run(appName, opts);
     } catch (error) {
       logError(error instanceof Error ? error.message : String(error));
       process.exit(1);
     }
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts:184]: Addressed in 2a47389bb. SIGINT cancellation is now handled as a clean cancellation in the CLI action catch path (blank line + exit code 0), instead of logging an error and exiting 1.

File: packages/create-react-on-rails-app/src/index.ts [hunk 13]
```diff
@@ -44,6 +40,27 @@ function run(appName: string, rawOpts: Record<string, unknown>): void {
     process.exit(1);
   }
 
+  // When no mode flag is explicitly passed, prompt interactively (TTY only).
+  // Non-interactive environments (CI, pipes) fall back to standard mode.
+  const modeExplicit = rawOpts.pro !== undefined || rawOpts.rsc !== undefined;
+  if (!modeExplicit) {
+    if (process.stdin.isTTY && process.stdout.isTTY) {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts:48]: Addressed in 2a47389bb. Prompt gating now checks `process.stdin.isTTY` (interactive input capability) rather than requiring both stdin and stdout TTY.

File: packages/create-react-on-rails-app/src/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Addressed in 2a47389bb. `program.parseAsync()` now has a `.catch(...)` handler so Commander-level async rejections are surfaced via `logError` and exit with status 1.

[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Addressed in 2a47389bb. `program.parseAsync()` now has a `.catch(...)` handler so Commander-level async rejections are surfaced via `logError` and exit with status 1.

File: packages/create-react-on-rails-app/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Skipping this one for now as non-blocking readability polish. The behavior is covered by the current option declarations and tests, and we kept this PR focused on correctness fixes.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      }
+    });
+
+    console.log(chalk.bold('Select a setup mode:\n'));
+    for (const mode of MODES) {
+      const recommended = mode.key === DEFAULT_KEY ? chalk.cyan(' (recommended)') : '';
+      console.log(`  ${mode.key}. ${chalk.bold(mode.label.padEnd(10))} ${mode.desc}${recommended}`);
+    }
+    console.log('');
+
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:64]: Skipping this as written because it is a stale/over-broad duplicate. EOF close fallback is already handled; the remaining blocking issue was the SIGINT settlement race, which is fixed in 2a47389bb.

File: packages/create-react-on-rails-app/tests/prompt.test.ts [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+describe('promptForMode', () => {
+  let consoleLogSpy: jest.SpyInstance;
+  let mockRl: { question: jest.Mock; close: jest.Mock; on: jest.Mock; once: jest.Mock };
+  let eventHandlers: Record<string, ((...args: unknown[]) => void)[]>;
+
+  beforeEach(() => {
+    consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+    eventHandlers = {};
+    mockRl = {
+      question: jest.fn(),
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/prompt.test.ts:17]: No change needed here: the test mock already includes `on` (and `once`) in the current branch, so this concern is already addressed.

**Comment by justin808** [comment]
Address-review checkpoint for PR #3063.

Scan mode: default cutoff mode (no `check all reviews` override).
Scan window: 2026-04-04T02:43:12Z to 2026-04-04T03:06:14Z.
Action completed at: 2026-04-04T09:55:35Z.

### Mattered
- **MUST-FIX** `packages/create-react-on-rails-app/src/prompt.ts` SIGINT settlement race: fixed in commit `2a47389bb` by preventing close-fallback resolution on SIGINT and preserving deterministic cancellation behavior.
- **DISCUSS (applied by explicit user choice)** `packages/create-react-on-rails-app/src/index.ts` Ctrl+C cancellation now exits cleanly (status 0), prompt gating now keys off `stdin.isTTY`, and `program.parseAsync()` now has a `.catch(...)` for Commander-level rejections.
- Addressed review threads were replied to and resolved.

### Skipped
- `index.ts` mode-detection explanatory comment request: skipped as non-blocking readability polish (replied with rationale).
- `prompt.ts` broad hang report: skipped as stale/duplicate framing (replied with rationale; underlying SIGINT bug fixed).
- `prompt.test.ts` add `on` to mock: skipped because already present in current branch (replied with rationale).
- Per user instruction, skipped-item threads were replied to without resolution.

No follow-up issue was created in this run.

Future full-PR scans should start after this comment unless `check all reviews` is explicitly requested.

File: packages/create-react-on-rails-app/src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Addressed in 3e0c772d8: changed the guard to `process.stdin.isTTY && process.stdout.isTTY`. Also updated the non-TTY log message and help text to mention stdout redirection.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    rl.on('SIGINT', () => {
+      answered = true;
+      rl.close();
+      reject(new Error(PROMPT_CANCELLED_BY_SIGINT));
+    });
+
+    rl.once('close', () => {
+      if (!answered) {
+        resolve({ pro: false, rsc: true });
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts]: Addressed in 3e0c772d8: EOF/Ctrl+D now rejects instead of silently defaulting to RSC. The constant was renamed from `PROMPT_CANCELLED_BY_SIGINT` to `PROMPT_CANCELLED` since it covers both SIGINT and EOF cancellation paths.

File: packages/create-react-on-rails-app/tests/prompt.test.ts [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+  it('displays mode options to the user', async () => {
+    mockRl.question.mockImplementation((_prompt: string, cb: (answer: string) => void) => cb('3'));
+
+    await promptForMode();
+
+    expect(consoleLogSpy).toHaveBeenCalledWith(expect.stringContaining('Standard'));
+    expect(consoleLogSpy).toHaveBeenCalledWith(expect.stringContaining('Pro'));
+    expect(consoleLogSpy).toHaveBeenCalledWith(expect.stringContaining('RSC'));
+  });
+});
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/prompt.test.ts:112]: Addressed in 3e0c772d8: added `tests/index-tty.test.ts` with 4 integration tests covering: TTY + no flags calls promptForMode, stdin not TTY skips prompt, stdout not TTY skips prompt, and explicit --rsc skips prompt.

File: packages/create-react-on-rails-app/src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Duplicate of another thread — addressed in 3e0c772d8 by gating on both `process.stdin.isTTY && process.stdout.isTTY`.

[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Duplicate of another thread — addressed in 3e0c772d8 by gating on both `process.stdin.isTTY && process.stdout.isTTY`.

File: packages/create-react-on-rails-app/src/prompt.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:64]: This was already addressed in a prior commit (2a47389bb) — the current code has both SIGINT and close handlers that prevent the Promise from hanging. No further changes needed.

File: packages/create-react-on-rails-app/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: The existing comment on lines 43-44 already explains the intent. Adding a more specific note about Commander.js internals would be a documentation nit — the undefined-vs-false behavior is standard Commander and unlikely to change without a major version bump.

File: packages/create-react-on-rails-app/tests/prompt.test.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/prompt.test.ts:17]: Already addressed in a prior commit — the mock now has `on: jest.fn()` and `once: jest.fn()` with proper event handler wiring.

File: packages/create-react-on-rails-app/src/index.ts [hunk 18]
```diff
@@ -145,13 +165,20 @@ The generated app includes one git commit per logical setup step.`,
 
 Documentation: https://reactonrails.com/docs/`,
   )
-  .action((appName: string, opts: Record<string, unknown>) => {
+  .action(async (appName: string, opts: Record<string, unknown>) => {
     try {
-      run(appName, opts);
+      await run(appName, opts);
     } catch (error) {
+      if (error instanceof Error && error.message === PROMPT_CANCELLED_BY_SIGINT) {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: The string comparison uses a shared constant (`PROMPT_CANCELLED`) so accidental collisions are not a realistic concern. A custom error class would add complexity without a practical benefit here.

File: packages/create-react-on-rails-app/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts]: Duplicate of another thread. The existing comment on lines 43-44 explains the intent sufficiently. The undefined-vs-false detection is standard Commander.js behavior.

File: packages/create-react-on-rails-app/src/index.ts [hunk 19]
```diff
@@ -145,13 +165,20 @@ The generated app includes one git commit per logical setup step.`,
...
+        console.log('');
+        process.exit(0);
+      }
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts:181]: Low severity — `process.exit()` terminates the process in production, so the subsequent lines are unreachable. The test suite does not mock `process.exit` in a way that would trigger this, so no change needed.

**Comment by justin808** [comment]
## Address-review summary (3e0c772d8)

Addressed 3 items from review, replied to 8 skipped items, resolved all 11 unresolved threads.

### Fixed

1. **stdout TTY check** (`index.ts:47`) — Interactive prompt now gates on both `process.stdin.isTTY && process.stdout.isTTY`. When stdout is redirected (e.g. `> setup.log`), the CLI falls back to standard mode instead of blocking with an invisible prompt. Help text updated to mention stdout redirection.

2. **EOF/Ctrl+D rejection** (`prompt.ts:54`) — EOF now rejects the prompt (like SIGINT) instead of silently defaulting to RSC. Users who hit Ctrl+D to cancel no longer get an unexpected RSC app. Constant renamed from `PROMPT_CANCELLED_BY_SIGINT` to `PROMPT_CANCELLED`.

3. **TTY integration tests** (`tests/index-tty.test.ts`) — Added 4 integration tests for the TTY-detection branching in `run()`:
   - TTY + no flags → calls `promptForMode`
   - stdin not TTY → skips prompt, logs non-interactive message
   - stdout not TTY → skips prompt, logs non-interactive message
   - TTY + explicit `--rsc` → skips prompt

### Skipped (with rationale replies posted)

- Promise hang on Ctrl+C/EOF — already fixed in prior commit
- Commander.js contract comments (×2) — existing comment sufficient
- Test mock `on` method — already fixed in prior commit
- Fragile error message matching — shared constant prevents collision
- SIGINT catch fallthrough — `process.exit()` terminates in production

All 91 tests passing.

File: packages/create-react-on-rails-app/tests/index-tty.test.ts [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+function loadIndex(): Promise<void> {
+  return new Promise((resolve, reject) => {
+    jest.isolateModules(() => {
+      try {
+        // The module calls program.parseAsync() at the top level. We let it
+        // run, then settle on the next tick so the async action completes.
+        require('../src/index');
+        // Give the async action time to execute
+        setTimeout(resolve, 50);
+      } catch (err) {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/tests/index-tty.test.ts]: Addressed in bf4e81cd7. Replaced `setTimeout(resolve, 50)` with an exported `ready` promise that awaits the actual `parseAsync()` completion. Tests now await `mod.ready` directly, eliminating the timing dependency entirely.

File: docs/oss/getting-started/create-react-on-rails-app.md [hunk 21]
```diff
@@ -16,49 +16,54 @@ The generated home page links to the example pages, the key files React on Rails
...
 npx create-react-on-rails-app my-app --package-manager pnpm
 
-# Combine options
-npx create-react-on-rails-app my-app --rspack --package-manager pnpm
-
 # Combine RSC with Rspack
 npx create-react-on-rails-app my-app --rspack --rsc
 ```
 
 ### All Options
```
[the remark this answers is not in the document]
Comment (justin808) [reply docs/oss/getting-started/create-react-on-rails-app.md:61]: Addressed in bf4e81cd7. Reverted the Default column for `--pro` and `--rsc` back to `false` and added a prose note below the table explaining the TTY prompting behavior. Also added `--standard` to the table.

File: packages/create-react-on-rails-app/src/index.ts [hunk 22]
```diff
@@ -44,6 +40,29 @@ function run(appName: string, rawOpts: Record<string, unknown>): void {
     process.exit(1);
   }
 
+  // When no mode flag is explicitly passed, prompt interactively (TTY only).
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts:43]: Addressed in bf4e81cd7. Added `--standard` flag so all three modes are equally first-class from the CLI. Scripts can now use `npx create-react-on-rails-app my-app --standard` to skip the prompt, just like `--pro` and `--rsc`. Added a corresponding TTY integration test.

File: packages/create-react-on-rails-app/src/index.ts [hunk 23]
```diff
@@ -145,13 +168,20 @@ The generated app includes one git commit per logical setup step.`,
...
   )
-  .action((appName: string, opts: Record<string, unknown>) => {
+  .action(async (appName: string, opts: Record<string, unknown>) => {
     try {
-      run(appName, opts);
+      await run(appName, opts);
     } catch (error) {
+      if (error instanceof Error && error.message === PROMPT_CANCELLED) {
+        console.log('');
+        process.exit(0);
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/index.ts:180]: This behavior was explicitly discussed and decided in a previous review cycle — user-initiated cancellation (Ctrl+C / Ctrl+D) exits with status 0. The rationale: cancellation is a deliberate user action, not an error, and many CLI tools (e.g. npm init) follow the same convention. The prompt module rejects correctly; the caller catches it and exits cleanly.

File: packages/create-react-on-rails-app/src/prompt.ts [hunk 24]
```diff
@@ -0,0 +1,78 @@
...
+    desc: 'React Server Components (requires react_on_rails_pro)',
+    pro: false,
+    rsc: true,
+  },
+] as const;
+
+const DEFAULT_KEY = '3';
+
+export function promptForMode(): Promise<ModeChoice> {
+  return new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (justin808) [reply packages/create-react-on-rails-app/src/prompt.ts:38]: Declining this one — adding a 5-minute timeout introduces complexity for an edge case that essentially never occurs in practice (a pseudo-TTY that reports isTTY but never sends EOF or SIGINT). The existing SIGINT and EOF handlers already cover all realistic terminal scenarios. If this becomes a real problem it can be added later.

**Comment by justin808** [comment]
Address-review checkpoint for PR #3063.

Scan mode: default cutoff mode (no `check all reviews` override).
Scan window: 2026-04-04T09:55:35Z to 2026-04-05T08:10:00Z.
Action completed at: 2026-04-05T08:11:23Z.

### Fixed (3 items, commit bf4e81cd7)

1. **`index-tty.test.ts:44-48`** — Replaced flaky `setTimeout(resolve, 50)` with an exported `ready` promise that awaits the actual `parseAsync()` completion. Tests now await `mod.ready` directly, eliminating the timing dependency.
2. **`create-react-on-rails-app.md:58`** — Reverted docs Default column from `prompted (TTY)` to `false` for `--pro`/`--rsc`. Added a prose note below the table explaining TTY prompting behavior.
3. **`index.ts:43`** — Added `--standard` flag so all three modes (standard/pro/rsc) are equally first-class from the CLI. Scripts can skip the prompt for standard mode. Added corresponding TTY integration test.

### Skipped (2 items, rationale replies posted)

4. **`index.ts:175-177`** Exit with non-zero on cancel — previously decided: user-initiated cancellation exits 0 by explicit choice.
5. **`prompt.ts:36-38`** Add 5-minute prompt timeout — speculative edge case for pseudo-TTYs that never send EOF/SIGINT. Existing handlers cover all realistic scenarios.

All 5 unresolved threads addressed: 3 resolved with code changes, 2 replied to with rationale.
92 tests passing.

Future full-PR scans should start after this comment unless `check all reviews` is explicitly requested.
